watchtower-plugin/src/retrier.rs

```diff
@@ -93,93 +107,153 @@ impl Retrier {
         }
     }
 
-    async fn add_appointment(&self, tower_id: TowerId) -> Result<(), Error<&'static str>> {
+    /// Adds an appointment to pending for a given tower.
+    ///
+    /// If the tower is not currently being retried, a new entry for it is created, otherwise, the data is appended to the existing entry.
+    fn add_pending_appointment(&self, tower_id: TowerId, locator: Locator) {
+        let mut pending_appointments = self.pending_appointments.lock().unwrap();
+        if let std::collections::hash_map::Entry::Vacant(e) = pending_appointments.entry(tower_id) {
+            self.wt_client
+                .lock()
+                .unwrap()
+                .set_tower_status(tower_id, crate::TowerStatus::TemporaryUnreachable);
```

Comment:
nit: Whichever function asked for the tower to be retried would have already set it to `TemporaryUnreachable`.